### ISSUE TITLE
False negative on pattern match when file name has non-English characters

### ISSUE THREAD

**Issue by webmaster-edlevo** [touchifyapp/vsts-changed-files#26]
We came across issue when extension returns "false" even tough path which is under patterns we have given. Details as below.

Filter rule => Modules/MyRegistry/Services/Host/My.Host/**
Changed File => Modules/MyRegistry/Services/Host/My.Host/Scripts/Behörighetsansvarig.imp

We expect extension to return true as file is under the path given in Filter rule but it's returning false. Logs as below.

`>> Changed files:
	 - "Modules/MyRegistry/Services/Host/My.Host/Scripts/Beh\303\266righetsansvarig_Sthlm.imp"
> Filtering files using glob rules
>> CODECHANGED: No change detected since last succeeded build, setting "CodeChanged" to "false"`

We can see two problems here.

1. Plugin is not handling non-English characters in file name. Original file name is _Behörighetsansvarig.imp_ but extention is considering it as _Beh\303\266righetsansvarig_Sthlm.imp_.
2. Pugin should probably use [partial pattern match](https://github.com/isaacs/minimatch#partial) instead of following. [Reference]( https://github.com/touchifyapp/vsts-changed-files/blob/4647ba2e51691a1922317900b9277b245ed3d3d1/changed-files/lib/matcher.ts#L3)

![image](https://user-images.githubusercontent.com/103247264/162722027-a2d93fe6-21f6-4b60-9201-17f9fb440b87.png)

**Comment by iamrahul127** [comment]
@SomaticIT Could please you check https://stackblitz.com/edit/typescript-1xz4ij?file=index.ts? I am not expert in TS so not creating PR but look like small POC mentioned above can be solution.

**Comment by SomaticIT** [comment]
Hello @iamrahul127,

I checked your stackblitz. However the error you have is not related to the error you encounter in `vsts-changed-files`.

The error was caused by `git log` which surrounds results that contains unicode characters with quotes.

This means this path `Modules/MyRegistry/Services/Host/My.Host/Scripts/Behörighetsansvarig.imp` becomes `"Modules/MyRegistry/Services/Host/My.Host/Scripts/Beh\303\266righetsansvarig_Sthlm.imp"`.

We implemented a fix to strip this quotes and it seems to work great.
It will be published as part of the `v1.2.2` release.
